vm-setup/roles/libvirt/tasks/vm_setup_tasks.yml

```diff
@@ -9,6 +9,24 @@
     state: directory
     mode: 0755
 
+- block:
+    - name: Detect virtualization if libvirt_domain_type is not provided
+      command: systemd-detect-virt
+      ignore_errors: true
+      become: true
+      register: virt_result
+
+    - name: Default to qemu if inside a VM
```

Comment:
why defaulting qemu inside a VM ? we used KVM as default ! can explain the impact of this when using generic ubuntu VM ?

Reply:
pure kvm won't always work in a VM, nested virtualization needs to be enabled as well as hardware pass-through
for example, in the ironic upstream CI that is openstack based, we use qemu on the VMs because of security concerns of some of the public cloud providers where the CI is configured

Reply:
Could qemu demand more resources ?

Reply:
not really, but I haven't seen the same in ironic upstream CI
to make this pass I'll introduce a variable for the kvm/qemu switch so we can set that when there is no kvm support and make it work fine in ironic CI
thanks for the help!

Reply:
I'm leaving this autoset option here for future cases or if we decide to set LIBVIRT_DOMAIN_TYPE to null

Reply:
When using kvm the nested virtualization can be enabled from the host or from the infrastructure that virtualize the host ?

Reply:
nested virtualization is enabled on the host, therefore whoever manages the infrastructure where the hosts run can decide to enable it or not